pyfr/solvers/navstokes/elements.py

```diff
@@ -8,6 +10,28 @@ class NavierStokesElements(BaseFluidElements, BaseAdvectionDiffusionElements):
     # Use the density field for shock sensing
     shockvar = 'rho'
 
+    @staticmethod
+    def grad_con_to_pri(cons, grad_cons, cfg):
+        rho = cons[0]
+        rhouvw = cons[1:-1]
+        grad_rho, *grad_rhouvw, grad_E = grad_cons
+        
+        # Divide momentum components by rho
+        vs = [rhov/rho for rhov in cons[1:-1]]
+
+        # Velocity gradients
+        # ∇(\vec{u}) = 1/\rho*[∇(\rho\vec{u}) - \vec{u} \otimes ∇(\rho)]
+        grad_v = [(grad_rhov - v*grad_rho)/rho for grad_rhov, v in zip(grad_cons[1:-1], vs)]
+
+        # Pressure gradient
+        # ∇p = (gamma - 1)*[∇E - 1/2*(\vec{u}·∇(\rho\vec{u}) - \rho\vec{u}·∇(\vec{u}))]
+        gamma = cfg.getfloat('constants', 'gamma')
+        grad_p = grad_E - 0.5*(np.einsum('ijk, iljk -> ljk', vs, grad_rhouvw) \
```

Comment:
No need for \, see if the + will fit on the line above.

Reply:
Not sure if I understood it correctly, let me know if the new version works for you